apis/Google.Cloud.Language.V1/Google.Cloud.Language.V1.Snippets/RpcExceptionExtensions.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Google.Rpc;
+using Grpc.Core;
+using System.Linq;
+
+using Status = Google.Rpc.Status;
+
+namespace Google.Api.Gax.Grpc
+{
+    /// <summary>
+    /// Utility extension methods to make it easier to retrieve extended error information from an <see cref="RpcException"/>.
+    /// </summary>
+    public static class RpcExceptionExtensions
+    {
+        public static Status GetRpcStatus(this RpcException ex) =>
+            DecodeTrailer(GetTrailer(ex, "grpc-status-details-bin"), Status.Parser);
+
+        // TODO: This attempts to find the BadRequest in two different trailers. Is
+        // that appropriate?
```

Comment:
"The error_details will be sent on the wire as gRPC metadata with key grpc-status-details-bin" [link](https://g3doc.corp.google.com/net/grpc/g3doc/cpp/status_mapping.md?cl=head#details-on-conversion)
In Typescript and go is based on `grpc-status-details-bin` since it is documented in previous.  However, I tried ask a [YAQS question](https://yaqs.corp.google.com/eng/q/1466717725127606272) about this for gRPC team. But haven't received any answer yet.

Reply:
If we're absolutely confident that if it's available anywhere, it'll *always* be in grpc-status-details-bin, then that would definitely make things simpler.

Reply:
Loop around with gRPC team, and confirmed that the `grpc-status-details-bin` is the source of truth. Update the answer in YAQS question.

Reply:
Righto. Will update this on Monday. (I'm on vacation tomorrow.)